server/service/src/purchase_order/delete/mod.rs

```diff
@@ -0,0 +1,182 @@
+use repository::{PurchaseOrderRowRepository, RepositoryError, TransactionError};
+
+pub mod validate;
+
+use validate::validate;
+
+use crate::{
+    purchase_order::common::get_lines_for_purchase_order,
+    purchase_order_line::delete::{delete_purchase_order_line, DeletePurchaseOrderLineError},
+    service_provider::ServiceContext,
+    WithDBError,
+};
+
+type OutError = DeletePurchaseOrderError;
+
+pub fn delete_purchase_order(
+    ctx: &ServiceContext,
+    store_id: &str,
+    id: String,
+) -> Result<String, DeletePurchaseOrderError> {
+    let purchase_order_id =
+        ctx.connection
+            .transaction_sync(|connection| {
+                validate(&id, store_id, connection)?;
+
+                let lines = get_lines_for_purchase_order(connection, &id)?;
+                for line in lines {
+                    delete_purchase_order_line(ctx, line.purchase_order_line_row.id.clone())
+                        .map_err(|error| OutError::LineDeleteError {
+                            line_id: line.purchase_order_line_row.id,
+                            error,
+                        })?;
+                }
```

Comment:
My understanding was that we should only delete empty purchase orders regardless of status. Is that right? Or should you be able to delete new POs along with all of their lines?